Gems/Multiplayer/Code/Tests/NetworkEntityTests.cpp

```diff
@@ -0,0 +1,373 @@
+        m_entityReplicationManager->AddDeferredRpcMessage(message2);
+        AZ_TEST_START_TRACE_SUPPRESSION;
+        m_entityReplicationManager->SendUpdates();
+        AZ_TEST_STOP_TRACE_SUPPRESSION(3);
+
+        m_entityReplicationManager->SetReplicationWindow(AZStd::make_unique<NullReplicationWindow>(m_mockConnection.get()));
+        m_entityReplicationManager->AddDeferredRpcMessage(message);
+        m_entityReplicationManager->AddDeferredRpcMessage(message2);
+        AZ_TEST_START_TRACE_SUPPRESSION;
+        m_entityReplicationManager->SendUpdates();
+        AZ_TEST_STOP_TRACE_SUPPRESSION(0);
+        NetworkEntityRpcVector rpcVector;
+        rpcVector.push_back(message);
+        m_entityReplicationManager->HandleEntityRpcMessages(m_mockConnection.get(), rpcVector);
+    }
+
+    TEST_F(MultiplayerNetworkEntityTests, TestNetworkEntityUpdateMessage)
+    {
+        ConstNetworkEntityHandle handle(m_root->m_entity.get(), m_networkEntityManager->GetNetworkEntityTracker());
+
+        NetworkEntityUpdateMessage message = NetworkEntityUpdateMessage();
+
+        // Test getters
+        EXPECT_EQ(message.GetNetworkRole(), NetEntityRole::InvalidRole);
+        EXPECT_EQ(message.GetData(), nullptr);
+        EXPECT_EQ(message.GetEntityId(), InvalidNetEntityId);
+        EXPECT_FALSE(message.GetHasValidPrefabId());
+        EXPECT_FALSE(message.GetIsDelete());
+        EXPECT_TRUE(message.GetPrefabEntityId().m_prefabName.IsEmpty());
+        EXPECT_FALSE(message.GetWasMigrated());
+
+        // Test setters
+        PrefabEntityId prefabId;
+        prefabId.m_entityOffset = 0;
+        prefabId.m_prefabName = AZ::Name("Test");
+        message.SetPrefabEntityId(prefabId);
+        EXPECT_NE(message.GetPrefabEntityId().m_prefabName.GetCStr(), "");
+        message.SetData(message.ModifyData());
+        EXPECT_NE(message.GetData(), nullptr);
+
+        AzNetworking::StringifySerializer serializer;
+        EXPECT_TRUE(message.Serialize(serializer));
+        EXPECT_EQ(message.GetEstimatedSerializeSize(), 17);
+
+        // Test ctors, assignment and comparisons (const and non const versions)
+        message = NetworkEntityUpdateMessage(m_root->m_netId, false);
+        EXPECT_EQ(m_root->m_netId, message.GetEntityId());
+        message = NetworkEntityUpdateMessage(NetEntityRole::Authority, m_root->m_netId);
+        EXPECT_EQ(message.GetNetworkRole(), NetEntityRole::Authority);
+        message = NetworkEntityUpdateMessage(NetEntityRole::Authority, m_root->m_netId, prefabId);
+        AzNetworking::PacketEncodingBuffer buffer;
+        EXPECT_NE(message.GetPrefabEntityId().m_prefabName.GetCStr(), "");
+        message.SetData(buffer);
+        NetworkEntityUpdateMessage message2(AZStd::move(message));
+        EXPECT_EQ(message, message2);
+        EXPECT_TRUE(m_root->m_replicator->GetPropertyPublisher()->PrepareSerialization());
+        const NetworkEntityUpdateMessage constMessage = m_root->m_replicator->GenerateUpdatePacket();
```

Comment:
Can this be used to trigger updates in network properties?

Reply:
Most actual network communication is pretty hard to do here unfortunately. We can basically exercise basic functionality and steps that prepare/serialize/deserialize.  In general, I think that's a pretty good goal for network based unit tests.